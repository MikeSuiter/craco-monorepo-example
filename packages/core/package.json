{
  "name": "@craco/core",
  "version": "0.1.0",
  "private": true,
  "main": "src",
  "module": "src",
  "dependencies": {
    "@craco/btns": "^0.1.0"
  },
  "scripts": {
    "start": "craco start",
    "build": "craco build",
    "test": "craco test",
    "eject": "craco eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
