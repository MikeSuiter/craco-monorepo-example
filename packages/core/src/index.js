import Layout from './components/Layout';

export { Layout };
