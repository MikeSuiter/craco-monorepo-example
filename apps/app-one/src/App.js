import logo from './logo.svg';
import './App.css';
import LocalButton from './components/LocalButton';
import { Layout } from '@craco/core';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a className="App-link" href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
          Learn React
        </a>
        <p>
          <LocalButton label="Local Button" />
          <Layout />
        </p>
      </header>
    </div>
  );
}

export default App;
