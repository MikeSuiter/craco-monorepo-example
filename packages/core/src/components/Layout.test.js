import { render, screen } from '@testing-library/react';
import { Layout } from '@craco/core/src';

describe('Layout tests', () => {
  test('renders correctly', () => {
    const cmp = render(<Layout />);
    expect(cmp).toMatchSnapshot();
  });

  test('renders blue button', () => {
    render(<Layout />);
    const linkElement = screen.getByText(/remote blue/i);
    expect(linkElement).toBeInTheDocument();
  });

  test('renders pink button', () => {
    render(<Layout />);
    const linkElement = screen.getByText(/remote pink/i);
    expect(linkElement).toBeInTheDocument();
  });
});
