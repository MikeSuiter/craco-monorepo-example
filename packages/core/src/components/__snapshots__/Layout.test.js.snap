// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        <button
          data-testid="core-button"
          style="background-color: darkblue; color: white;"
        >
          Remote Blue
        </button>
        <button
          data-testid="core-button"
          style="background-color: deeppink; color: white;"
        >
          Remote Pink
        </button>
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      <button
        data-testid="core-button"
        style="background-color: darkblue; color: white;"
      >
        Remote Blue
      </button>
      <button
        data-testid="core-button"
        style="background-color: deeppink; color: white;"
      >
        Remote Pink
      </button>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
